pkg/server/cloud/aws/instances.go

```diff
@@ -231,6 +231,18 @@ func bootImageSpecToDescribeImagesInput(spec cloud.BootImageSpec) *ec2.DescribeI
 	return input
 }
 
+func getRootDeviceVolumeSize(blockDevices []*ec2.BlockDeviceMapping, rootDeviceName string) int32 {
+	var rootDiskSize int32
+	for _, blockDevice := range blockDevices {
+		if *blockDevice.DeviceName == rootDeviceName {
```

Comment:
When dealing with AWS data structures and values, I've found it safer to follow the assumption: "Almost everything can have bad data". If a cloud provider pushes out a change to their API that has a bug (rare but can happen) and that bug returns a nil value for `DeviceName` or `blockDevice.Ebs`, Kip will crash. Also there can be times where something you thought "This can't ever be nil" turns out to be nil. E.g: https://github.com/elotl/kip/pull/173

It kinda stinks but it's better to be defensive against these types of errors.  I would suggest the following change to protect against nil values:

```go
		if aws.StringValue(blockDevice.DeviceName) == rootDeviceName &&
			blockDevice.Ebs != nil {
			rootDiskSize = aws.Int32Value(blockDevice.Ebs.VolumeSize)
			break
		}
```

Reply:
Makes sense, I've added check if Ebs is nil. I kept `rootDiskSize = int32(blockDevice.Ebs.VolumeSize)` as originally `blockDevice.Ebs.VolumeSize` is int64, so we have to convert it.

Reply:
In that case the same point stands but with different data types, I would suggest `rootDiskSize = int32(aws.Int64Value(blockDevice.Ebs.VolumeSize))`.

What we've really been strongly recommending: it's good coding practice to use AWS's unpacking functions when dealing with their datastructures.  Directly dereferencing data from those structures without testing for nil comes with risks.